workers.tf

```
@@ -73,6 +73,16 @@ resource "aws_autoscaling_group" "workers" {
     "termination_policies",
     local.workers_group_defaults["termination_policies"]
   )
+  default_cooldown = lookup(
+    var.worker_groups_launch_template[count.index],
+    "default_cooldown",
+    local.workers_group_defaults["default_cooldown"]
+  )
+  health_check_grace_period = lookup(
+    var.worker_groups_launch_template[count.index],
```

You have copy pasted something wrong here:

```suggestion
    var.worker_groups[count.index],
```